mypy/stubgenc.py

```diff
@@ -307,29 +302,38 @@ def is_skipped_attribute(attr: str) -> bool:
                     '__weakref__')  # For pickling
 
 
-def infer_method_sig(name: str) -> str:
+def infer_method_sig(name: str) -> List[TypedArgSig]:
     if name.startswith('__') and name.endswith('__'):
         name = name[2:-2]
         if name in ('hash', 'iter', 'next', 'sizeof', 'copy', 'deepcopy', 'reduce', 'getinitargs',
                     'int', 'float', 'trunc', 'complex', 'bool'):
-            return '()'
+            return []
         if name == 'getitem':
-            return '(index)'
+            return [TypedArgSig(name='index', type=None, default=False)]
         if name == 'setitem':
-            return '(index, object)'
+            return [
+                TypedArgSig(name='index', type=None, default=False),
+                TypedArgSig(name='object', type=None, default=False)
+            ]
         if name in ('delattr', 'getattr'):
-            return '(name)'
+            return [TypedArgSig(name='name', type=None, default=False)]
         if name == 'setattr':
-            return '(name, value)'
+            return [
+                TypedArgSig(name='name', type=None, default=False),
+                TypedArgSig(name='value', type=None, default=False)
+            ]
```

Comment:
For multiline we use this style:
```python
return [TypedArgSig(name='name', type=None, default=False),
        TypedArgSig(name='value', type=None, default=False)]
```
(also many of these will not need to be multiline if you implement the suggestion above).